app/src/main/java/com/babylon/wallet/android/domain/utils/AccountLockersObserver.kt

```diff
@@ -0,0 +1,134 @@
+package com.babylon.wallet.android.domain.utils
+
+import com.babylon.wallet.android.data.repository.locker.AccountLockersRepository
+import com.babylon.wallet.android.di.coroutines.ApplicationScope
+import com.babylon.wallet.android.di.coroutines.DefaultDispatcher
+import com.babylon.wallet.android.domain.model.locker.AccountLockerDeposit
+import com.babylon.wallet.android.domain.usecases.GetDAppsUseCase
+import com.radixdlt.sargon.AccountAddress
+import com.radixdlt.sargon.AuthorizedDapp
+import com.radixdlt.sargon.AuthorizedDappPreferenceDeposits
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.combine
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.flow
+import kotlinx.coroutines.flow.flowOn
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.launch
+import rdx.works.core.domain.DApp
+import rdx.works.profile.data.repository.DAppConnectionRepository
+import timber.log.Timber
+import javax.inject.Inject
+import javax.inject.Singleton
+import kotlin.time.Duration.Companion.minutes
+
+@Singleton
+class AccountLockersObserver @Inject constructor(
+    private val getDAppsUseCase: GetDAppsUseCase,
+    private val dAppConnectionRepository: DAppConnectionRepository,
+    private val accountLockersRepository: AccountLockersRepository,
+    @ApplicationScope private val appScope: CoroutineScope,
+    @DefaultDispatcher private val defaultDispatcher: CoroutineDispatcher
+) {
+
+    private var monitoringJob: Job? = null
+    private val depositsByAccount = MutableSharedFlow<Map<AccountAddress, List<AccountLockerDeposit>>>(1)
```

Comment:
isn't that StateFlow since replay is 1? or you didn't want to give an empty map as initial value?

Reply:
Exactly, that's the reason.